### ISSUE TITLE
Track a remote branch for an MSC

### ISSUE THREAD

**Issue by ShadowJonathan** [matrix-org/mscbot-python#21]
#### Is your feature request related to a problem? Please describe.
In situations where the original author may want to delegate an MSC, and where - due to security reasons, or likewise - the author is unable to grant the delegate author access to the PR's origin repository, it may want to have the PR branch be updated from a remote branch instead, such as the delegate author's own fork.

In other situations, an organisation may want to - for legal reasons, or likewise - develop the commits on their own git server, but have them be fetched to the MSC's PR branch.

#### Describe the solution you'd like.
For these problems, I'd like the mscbot to be able to be set up to watch a remote branch. Be it on github, or on any compatible git server.

The original author of the PR has to setup the watch, via `@mscbot watch <repository URL> <branch>`.

After this, the MSC bot will only try to fetch commits from that branch via a `git pull`-like mechanism. That is to say; it will not force-push the PR branch with updates from the remote branch, it will only append new commits that were present in the remote branch, but not in the PR branch.

In other words, the MSC bot will only [fast-forward](https://stackoverflow.com/a/29673993/8700553) the PR branch to the remote branch.

**Comment by ShadowJonathan** [comment]
Alternatively to the `watch` command, the original author could add `Watch: <repo URL> <branch>` to the description instead.

This'd enable the mscbot to be more stateless, so I think that might be a better solution.

**Comment by anoadragon453** [comment]
It would be easiest to watch branches via github webhook (or if other platforms are supported), otherwise you would need to poll potentially hundreds of git repos, which is a lot more than what the bot is doing today.

**Comment by ShadowJonathan** [comment]
Probably a hybrid; Webhook the repository servers that support it, and poll (hourly or 6-hourly) the remaining servers.
